registry/storage/driver/middleware/cloudfront/middleware.go

```diff
@@ -114,8 +130,66 @@ type S3BucketKeyer interface {
 	S3BucketKey(path string) string
 }
 
-// Resolve returns an http.Handler which can serve the contents of the given
-// Layer, or an error if not supported by the storagedriver.
+// parseIPFromRequest attempts to extract the ip address of the
+// client that made the request
+func parseIPFromRequest(ctx context.Context) (net.IP, error) {
+	request, err := dcontext.GetRequest(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ipStr := dcontext.RemoteIP(request)
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		return nil, fmt.Errorf("invalid ip address from requester: %s", ipStr)
+	}
+
+	return ip, nil
+}
+
+// Client facing behavior changed slightly with the rollout of the s3filter.
```

Comment:
Addressing client issues by dropping the headers on the server is a little problematic in that we can leak the authorization headers any ways, by a crafted request.

We've also taken the policy of _never_ introspecting the client version on the registry, as it can lead to confusing or buggy behavior.